app/controllers/api/restful_controller.rb

```diff
@@ -18,7 +20,8 @@ def destroy_action
   private
 
   def current_user
-    super || LoggedOutUser.new
+    token_user = User.find(doorkeeper_token.resource_owner_id) if doorkeeper_token
```

Comment:
thanks @gdpelican. I will investigate caching the doorkeeper and writing some basic testing